samples/snapshot/src/test/java/pubsublite/spark/SampleIntegrationTest.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.Invoker;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.spark_project.guava.collect.ImmutableList;
+
+public class SampleIntegrationTest {
+
+  private static final CloudRegion CLOUD_REGION = CloudRegion.of("us-central1");
+  CloudZone CLOUD_ZONE = CloudZone.of(CLOUD_REGION, 'b');
+  ProjectId PROEJCT_ID = ProjectId.of("pubsub-lite-load-tests");
+  TopicName TOPIC_NAME = TopicName.of("jiangmichael-sample-integration");
+  SubscriptionName SUBSCRIPTION_NAME =
+      SubscriptionName.of("sample-integration-sub-" + System.currentTimeMillis());
+  SubscriptionPath SUBSCIPTION_PATH =
+      SubscriptionPath.newBuilder()
+          .setProject(PROEJCT_ID)
+          .setLocation(CLOUD_ZONE)
+          .setName(SUBSCRIPTION_NAME)
+          .build();
+  String CLUSTER_NAME = "jiangmichael-psl-spark-connector-test-cluster-2";
+  String BUCKET_NAME = "jiangmichael-spark-sample-integration";
+  String WORKING_DIR = System.getProperty("user.dir").replace("/samples/snapshot", "");
+  String SAMPLE_JAR_NAME = "pubsublite-spark-snapshot-1.0.21.jar";
+  String CONNECTOR_JAR_NAME = "pubsublite-spark-sql-streaming-with-dependencies-0.1.0-SNAPSHOT.jar";
+  String SAMPLE_JAR_LOC =
+      String.format("%s/samples/snapshot/target/%s", WORKING_DIR, SAMPLE_JAR_NAME);
+  String CONNECTOR_JAR_LOC = String.format("%s/target/%s", WORKING_DIR, CONNECTOR_JAR_NAME);
+
+  private static void mavenPackage(String workingDir) throws MavenInvocationException {
+    InvocationRequest request = new DefaultInvocationRequest();
+    request.setPomFile(new File(workingDir + "/pom.xml"));
+    request.setGoals(ImmutableList.of("clean", "package", "-Dmaven.test.skip=true"));
+    Invoker invoker = new DefaultInvoker();
+    invoker.setMavenHome(new File("/opt/apache-maven-3.5.3"));
+    assertThat(invoker.execute(request).getExitCode()).isEqualTo(0);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    // Create a subscription
+    createSubscriptionExample(
+        CLOUD_REGION.value(),
+        CLOUD_ZONE.zoneId(),
+        PROEJCT_ID.value(),
+        TOPIC_NAME.value(),
+        SUBSCRIPTION_NAME.value());
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    // Cleanup the subscription
+    deleteSubscriptionExample(
+        CLOUD_REGION.value(), CLOUD_ZONE.zoneId(), PROEJCT_ID.value(), SUBSCRIPTION_NAME.value());
+  }
+
+  @Test
+  public void test() throws Exception {
+    // Maven package into jars
+    mavenPackage(WORKING_DIR);
+    mavenPackage(WORKING_DIR + "/samples");
+
+    // Upload to GCS
+    Storage storage =
+        StorageOptions.newBuilder().setProjectId(PROEJCT_ID.value()).build().getService();
+    BlobId blobId = BlobId.of(BUCKET_NAME, SAMPLE_JAR_NAME);
```

Comment:
Avoid parallel uploads from parallel tests.

```suggestion
    BlobId blobId = BlobId.of(BUCKET_NAME, suffix, SAMPLE_JAR_NAME);
```